File: Sample Applications/AppPortfolio App/AppPortfolioCS/AppDetailsPanel.cs
using System;
using System.Drawing;
using System.Drawing.Imaging;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.Layouts;
using Telerik.WinControls.Primitives;
using Telerik.WinControls.UI;

namespace AppPortfolio
{
    public class AppDetailsPanel : RadControl
    {
        private AppDetailsPanelElement panelElement = null;

        static AppDetailsPanel()
        {
            ThemeResolutionService.LoadPackageResource("AppPortfolio.Resources.AppPortfolioTheme.tssp");
        }

        public AppDetailsPanel()
        {
            this.Size = new Size(950, 450);
            this.Location = new Point(0, 50);
            this.ThemeName = "AppPortfolioTheme";
            this.KeyDown += new KeyEventHandler(AppDetailsPanel_KeyDown);
        }

        private void AppDetailsPanel_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.F5)
                this.PanelElement.PerformInitAnimation();
            else if (e.KeyCode == Keys.Escape)
                this.Hide();
        }

        public AppDetailsPanelElement PanelElement
        {
            get
            {
                return this.panelElement;
            }
        }

        public PortfolioButtonElement PortfolioButton
        {
            get
            {
                return this.panelElement.PortfolioButton;
            }
            set
            {
                this.panelElement.PortfolioButton = value;
            }
        }

        protected override void OnDoubleClick(EventArgs e)
        {
            base.OnDoubleClick(e);

            this.Hide();
        }

        protected override void CreateChildItems(RadElement parent)
        {
            this.BackgroundImageLayout = System.Windows.Forms.ImageLayout.Stretch;
            this.BackgroundImage = Properties.Resources.BackgroundItemBig;

            base.CreateChildItems(parent);

            this.panelElement = new AppDetailsPanelElement();
            parent.Children.Add(this.panelElement);
        }
    }

    public class AppDetailsPanelElement : RadPanelElement
    {
        private ImagePrimitive productImage = null;
        private TextPrimitive titleLabel = null;
        private TextPrimitive descriptionLabel = null;
        private RadButtonElement buttonElement = null;
        private RadButtonElement backButtonElement = null;

        private PortfolioButtonElement portfolioButton;

        public Image ProductImage
        {
            get
            {
                return this.productImage.Image;
            }
            set
            {
                this.productImage.Image = value;
            }
        }

        public string Title
        {
            get
            {
                return this.titleLabel.Text;
            }
            set
            {
                this.titleLabel.Text = value;
            }
        }

        public string Description
        {
            get
            {
                return this.descriptionLabel.Text;
            }
            set
            {
                this.descriptionLabel.Text = value;
            }
        }

        public PortfolioButtonElement PortfolioButton
        {
            get
            {
                return portfolioButton;
            }
            set
            {
                if (this.portfolioButton != value)
                {
                    this.portfolioButton = value;

                    if (this.portfolioButton != null)
                    {
                        this.productImage.Image = this.portfolioButton.GetProductImage();
                        this.titleLabel.Text = this.portfolioButton.ProductTitle;
                        this.descriptionLabel.Text = this.portfolioButton.ProductDescription;
                    }
                    else
                    {
                        this.productImage.Image = null;
                        this.titleLabel.Text = string.Empty;
                        this.descriptionLabel.Text = string.Empty;
                    }
                }
            }
        }

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            this.productImage = new ImagePrimitive();
            this.productImage.Image = Properties.Resources.Telerik;

            this.descriptionLabel = new TextPrimitive();
            this.descriptionLabel.ForeColor = Color.White;
            this.descriptionLabel.Font = new Font("Segoe UI", 10f);
            this.descriptionLabel.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias;
            this.descriptionLabel.StretchHorizontally = true;
            this.descriptionLabel.TextAlignment = ContentAlignment.TopLeft;
            this.descriptionLabel.MaxSize = new Size(450, 0);
            this.descriptionLabel.MinSize = new Size(0, 200);
            this.descriptionLabel.TextWrap = true;
            this.descriptionLabel.Text = "Description";

            this.titleLabel = new TextPrimitive();
            this.titleLabel.Margin = new Padding(0, 0, 0, 20);
            this.titleLabel.ForeColor = Color.Red;
            this.titleLabel.Font = new Font("Verdana", 20f, FontStyle.Bold);
            this.titleLabel.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias;
            this.titleLabel.Text = "Title";

            BoxLayout verticalButtonLayout = new BoxLayout();
            verticalButtonLayout.Orientation = Orientation.Horizontal;
            this.buttonElement = new RadButtonElement();
            this.buttonElement.StretchHorizontally = false;
            this.buttonElement.StretchVertically = false;
            this.buttonElement.Text = "SEE DEMO";
            this.buttonElement.Click += new EventHandler(buttonElement_Click);

            this.backButtonElement = new RadButtonElement();
            this.backButtonElement.StretchHorizontally = false;
            this.backButtonElement.StretchVertically = false;
            this.backButtonElement.Text = "BACK";
            this.backButtonElement.Click += new EventHandler(backButtonElement_Click);

            verticalButtonLayout.Children.Add(buttonElement);
            verticalButtonLayout.Children.Add(backButtonElement);

            BoxLayout verticalLayout = new BoxLayout();
            verticalLayout.Orientation = System.Windows.Forms.Orientation.Vertical;
            verticalLayout.Children.Add(this.titleLabel);
            verticalLayout.Children.Add(this.descriptionLabel);
            //  verticalLayout.Children.Add(this.buttonElement);
            //  verticalLayout.Children.Add(this.backButtonElement);
            verticalLayout.Children.Add(verticalButtonLayout);

            BoxLayout horizontalLayout = new BoxLayout();
            horizontalLayout.Margin = new Padding(20);
            horizontalLayout.StretchHorizontally = true;
            horizontalLayout.StretchVertically = true;
            horizontalLayout.Orientation = System.Windows.Forms.Orientation.Horizontal;

            BoxLayout.SetProportion(this.productImage, 1);
            BoxLayout.SetProportion(verticalLayout, 1);
            horizontalLayout.Children.Add(this.productImage);
            horizontalLayout.Children.Add(verticalLayout);

            this.Children.Add(horizontalLayout);
        }

        private void backButtonElement_Click(object sender, EventArgs e)
        {
            this.ElementTree.Control.Hide();
        }

        private void buttonElement_Click(object sender, EventArgs e)
        {
            if (this.portfolioButton != null)
            {
                this.PortfolioButton.ExecuteCommand();
            }
        }

        public void PerformInitAnimation()
        {
            AnimatedPropertySetting setting = new AnimatedPropertySetting(
                            RadElement.MarginProperty,
                            new Padding(0, 20, 0, 0),
                            new Padding(0, 0, 0, 0),
                            10,
                            40);

            AnimatedPropertySetting setting1 = new AnimatedPropertySetting(
                            VisualElement.OpacityProperty,
                            0d,
                            1d,
                            10,
                            40);

            setting.ApplyValue(this.productImage);
            setting.ApplyValue(this.descriptionLabel);

            setting1.ApplyValue(this.productImage);
            setting1.ApplyValue(this.descriptionLabel);
            setting1.ApplyValue(this.titleLabel);

            setting = new AnimatedPropertySetting(
                            RadElement.MarginProperty,
                            new Padding(0, 20, 0, 20),
                            new Padding(0, 0, 0, 20),
                            10,
                            40);

            setting.ApplyValue(this.titleLabel);
        }
    }
}

File: Sample Applications/AppPortfolio App/AppPortfolioVB/AppDetailsPanel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports Telerik.WinControls
Imports Telerik.WinControls.Layouts
Imports Telerik.WinControls.Primitives
Imports Telerik.WinControls.UI
Imports System.Drawing

Namespace AppPortfolioVB
    Public Class AppDetailsPanel
        Inherits RadControl
        Private panelElement_Renamed As AppDetailsPanelElement


        Shared Sub New()
            ThemeResolutionService.LoadPackageResource("AppPortfolioTheme.tssp")
        End Sub

        Public Sub New()
            Me.Size = New Size(950, 450)
            Me.ThemeName = "AppPortfolioTheme"
            AddHandler KeyDown, AddressOf AppDetailsPanel_KeyDown
        End Sub

        Private Sub AppDetailsPanel_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs)
            If e.KeyCode = Keys.F5 Then
                Me.PanelElement.PerformInitAnimation()
            ElseIf e.KeyCode = Keys.Escape Then
                Me.Hide()
            End If
        End Sub

        Public ReadOnly Property PanelElement() As AppDetailsPanelElement
            Get
                Return Me.panelElement_Renamed
            End Get
        End Property

        Public Property PortfolioButton() As PortfolioButtonElement
            Get
                Return Me.panelElement_Renamed.PortfolioButton
            End Get
            Set(value As PortfolioButtonElement)
                Me.panelElement_Renamed.PortfolioButton = Value
            End Set
        End Property

        Protected Overrides Sub OnDoubleClick(ByVal e As EventArgs)
            MyBase.OnDoubleClick(e)

            Me.Hide()
        End Sub

        Protected Overrides Sub CreateChildItems(ByVal parent As RadElement)
            Me.BackgroundImageLayout = System.Windows.Forms.ImageLayout.Stretch
            Me.BackgroundImage = Resources.BackgroundItemBig

            MyBase.CreateChildItems(parent)

            Me.panelElement_Renamed = New AppDetailsPanelElement()
            parent.Children.Add(Me.panelElement_Renamed)
        End Sub
    End Class

    Public Class AppDetailsPanelElement
        Inherits RadPanelElement
        Private productImage_Renamed As ImagePrimitive
        Private titleLabel As TextPrimitive
        Private descriptionLabel As TextPrimitive
        Private buttonElement As RadButtonElement
        Private backButtonElement As RadButtonElement

        Private portfolioButton_Renamed As PortfolioButtonElement

        Public Property ProductImage() As Image
            Get
                Return Me.productImage_Renamed.Image
            End Get
            Set(value As Image)
                Me.productImage_Renamed.Image = Value
            End Set
        End Property

        Public Property Title() As String
            Get
                Return Me.titleLabel.Text
            End Get
            Set(value As String)
                Me.titleLabel.Text = Value
            End Set
        End Property

        Public Property Description() As String
            Get
                Return Me.descriptionLabel.Text
            End Get
            Set(value As String)
                Me.descriptionLabel.Text = Value
            End Set
        End Property

        Public Property PortfolioButton() As PortfolioButtonElement
            Get
                Return portfolioButton_Renamed
            End Get
            Set(value As PortfolioButtonElement)
                If Not Me.portfolioButton_Renamed Is Value Then
                    Me.portfolioButton_Renamed = Value

                    If Not Me.portfolioButton_Renamed Is Nothing Then
                        Me.productImage_Renamed.Image = Me.portfolioButton_Renamed.GetProductImage()
                        Me.titleLabel.Text = Me.portfolioButton_Renamed.ProductTitle
                        Me.descriptionLabel.Text = Me.portfolioButton_Renamed.ProductDescription
                    Else
                        Me.productImage_Renamed.Image = Nothing
                        Me.titleLabel.Text = String.Empty
                        Me.descriptionLabel.Text = String.Empty
                    End If
                End If
            End Set
        End Property

        Protected Overrides Sub CreateChildElements()
            MyBase.CreateChildElements()

            Me.productImage_Renamed = New ImagePrimitive()
            Me.productImage_Renamed.Image = Resources.Telerik

            Me.descriptionLabel = New TextPrimitive()
            Me.descriptionLabel.ForeColor = Color.White
            Me.descriptionLabel.Font = New Font("Segoe UI", 10.0F)
            Me.descriptionLabel.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias
            Me.descriptionLabel.StretchHorizontally = True
            Me.descriptionLabel.TextAlignment = ContentAlignment.TopLeft
            Me.descriptionLabel.MaxSize = New Size(450, 0)
            Me.descriptionLabel.MinSize = New Size(0, 200)
            Me.descriptionLabel.TextWrap = True
            Me.descriptionLabel.Text = "Description"

            Me.titleLabel = New TextPrimitive()
            Me.titleLabel.Margin = New Padding(0, 0, 0, 20)
            Me.titleLabel.ForeColor = Color.Red
            Me.titleLabel.Font = New Font("Verdana", 20.0F, FontStyle.Bold)
            Me.titleLabel.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias
            Me.titleLabel.Text = "Title"

            Me.buttonElement = New RadButtonElement()
            Me.buttonElement.StretchHorizontally = False
            Me.buttonElement.StretchVertically = False
            Me.buttonElement.Text = "SEE DEMO"
            AddHandler buttonElement.Click, AddressOf buttonElement_Click

            Me.backButtonElement = New RadButtonElement()
            Me.backButtonElement.StretchHorizontally = False
            Me.backButtonElement.StretchVertically = False
            Me.backButtonElement.Text = "BACK"
            AddHandler Me.backButtonElement.Click, AddressOf backButtonElement_Click

            Dim verticalLayout As BoxLayout = New BoxLayout()
            verticalLayout.Orientation = System.Windows.Forms.Orientation.Vertical
            verticalLayout.Children.Add(Me.titleLabel)
            verticalLayout.Children.Add(Me.descriptionLabel)
            verticalLayout.Children.Add(Me.buttonElement)
            verticalLayout.Children.Add(Me.backButtonElement)

            Dim horizontalLayout As BoxLayout = New BoxLayout()
            horizontalLayout.Margin = New Padding(20)
            horizontalLayout.StretchHorizontally = True
            horizontalLayout.StretchVertically = True
            horizontalLayout.Orientation = System.Windows.Forms.Orientation.Horizontal
            BoxLayout.SetProportion(Me.productImage_Renamed, 1)
            BoxLayout.SetProportion(verticalLayout, 1)
            horizontalLayout.Children.Add(Me.productImage_Renamed)
            horizontalLayout.Children.Add(verticalLayout)

            Me.Children.Add(horizontalLayout)
        End Sub

        Private Sub backButtonElement_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.ElementTree.Control.Hide()
        End Sub

        Private Sub buttonElement_Click(ByVal sender As Object, ByVal e As EventArgs)
            If Not Me.portfolioButton_Renamed Is Nothing Then
                Me.PortfolioButton.ExecuteCommand()
            End If
        End Sub

        Public Sub PerformInitAnimation()
            Dim setting As AnimatedPropertySetting = New AnimatedPropertySetting(RadElement.MarginProperty, New Padding(0, 20, 0, 0), New Padding(0, 0, 0, 0), 10, 40)

            Dim setting1 As AnimatedPropertySetting = New AnimatedPropertySetting(VisualElement.OpacityProperty, 0.0R, 1.0R, 10, 40)

            setting.ApplyValue(Me.productImage_Renamed)
            setting.ApplyValue(Me.descriptionLabel)

            setting1.ApplyValue(Me.productImage_Renamed)
            setting1.ApplyValue(Me.descriptionLabel)
            setting1.ApplyValue(Me.titleLabel)

            setting = New AnimatedPropertySetting(RadElement.MarginProperty, New Padding(0, 20, 0, 20), New Padding(0, 0, 0, 20), 10, 40)

            setting.ApplyValue(Me.titleLabel)
        End Sub
    End Class
End Namespace
